ddtrace/internal/writer.py

```diff
@@ -239,7 +239,7 @@ def __init__(
                 }
             )
 
-        self._encoder = Encoder()
+        self._encoder = Encoder(buf_size=self._max_payload_size)
```

Comment:
this doesn't work with other encoders, only the custom encoder, need to sort this if we want to go in this direction.

Reply:
It would if we go in the direction of #2422: https://github.com/DataDog/dd-trace-py/blob/49e77b546acff7b2539e09858899351889714030/ddtrace/internal/_encoding.pyi#L11 😉

Reply:
yeah, I was wondering what would overlap with your work. Probably good to get that reviewed/merged in first before taking a pass like this.